### PR TITLE
HPCC-24636 Fix issues with missing service accounts

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Using kube-score, and minikube

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: helm/hpcc/templates/roxie.yaml [hunk 1]
```diff
@@ -161,6 +163,7 @@ spec:
         roxie-server: {{ $servername | quote }}
 {{- end }}
     spec:
+      serviceAccountName: "hpcc-default"
```
Comment (jakesmith) [inline helm/hpcc/templates/roxie.yaml:166]: trivial: should we be quoting all strings for consistency? e.g. here vs below in service-account.yaml name:

File: testing/helm/run.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-options="--set global.image.version=latest"
+options="--set global.image.version=someversion --set global.image.pullPolicy=Always"
```
Comment (jakesmith) [inline testing/helm/run.sh:3]: what is the significance of this change?
Did kubeval/kube-score complain re use of "latest" ? and IfNotPresent default?

**Comment by jakesmith** [comment]
@richardkchapman - 2 minor questions

File: testing/helm/run.sh (hunk 2 above)
Comment (richardkchapman) [reply testing/helm/run.sh:3]: Yes

File: helm/hpcc/templates/roxie.yaml (hunk 1 above)
Comment (richardkchapman) [reply helm/hpcc/templates/roxie.yaml:166]: Not sure I care enough to go through and change...

Comment (jakesmith) [reply helm/hpcc/templates/roxie.yaml:166]: fair enough.

File: testing/helm/run.sh (hunk 2 above)
Comment (jakesmith) [reply testing/helm/run.sh:3]: ok

**Review by jakesmith — APPROVED** [review]

**Comment by jakesmith** [comment]
@richardkchapman - looks good.

**Comment by richardkchapman** [comment]
@ghalliday

**Pull request merged**